File: tools/python-execute-eval/agent-solutions/claude-sonnet-4-6/stats-stdlib.py
import statistics


async def main(params):
    values = params["values"]
    mean_val = float(statistics.mean(values))
    median_val = float(statistics.median(values))
    pstdev_val = float(statistics.pstdev(values))
    return {
        "mean": round(mean_val, 6),
        "median": round(median_val, 6),
        "pstdev": round(pstdev_val, 6),
    }

File: tools/python-execute-eval/agent-solutions/claude-sonnet-4-6/stats-stdlib.mjs
import { kody } from 'kody:runtime'

export default async function main(params) {
	const values = params.values
	const n = values.length
	const mean = values.reduce((sum, v) => sum + v, 0) / n
	const sorted = [...values].sort((a, b) => a - b)
	const mid = Math.floor(sorted.length / 2)
	const median =
		sorted.length % 2 === 0 ? (sorted[mid - 1] + sorted[mid]) / 2 : sorted[mid]
	const variance = values.reduce((sum, v) => sum + Math.pow(v - mean, 2), 0) / n
	const pstdev = Math.sqrt(variance)
	const round6 = (x) => parseFloat(x.toFixed(6))
	return { mean: round6(mean), median: round6(median), pstdev: round6(pstdev) }
}
